forge/db/controllers/Device.js

```diff
@@ -254,11 +255,34 @@ module.exports = {
         const ids = idsDecoded.map(id => id && Array.isArray(id) ? id[0] : null).filter(id => id)
 
         // find all where id in ids
-        const devices = await app.db.models.Device.findAll({ where: { id: ids } })
+        const devices = await app.db.models.Device.findAll({
+            where: { id: ids },
+            include: {
+                model: app.db.models.Project,
+                attributes: ['id', 'ApplicationId']
+            }
+        })
         if (devices.length === 0) {
             throw new ControllerError('not_found', 'No devices found', 404)
         }
 
+        if (!user.admin && teamMembership && teamMembership.permissions?.applications) {
```

Comment:
One concern: both methods check if RBAC is enabled by checking if teamMembership & teamMembership.permissions are passed, but disabling RBAC on a team doesn’t cause the check to fail since `TeamMembership.permissions` aren’t cleared.

Instead of adding another argument, we could verify the feature is enabled at both the team and platform level, and only pass `teamMembership` if it is. This avoids extra arguments and ensures checks reflect the current state.

Reply:
Added feature flag checks